core/src/main/resources/org/apache/spark/ui/static/executorspage.js

```diff
@@ -770,5 +940,140 @@ $(document).ready(function () {
         }
       });
     });
+
+    var quantiles = "0,0.25,0.5,0.75,1.0";
+    $.getJSON(createRESTEndPointForExecutorsPeakMetricsSummariesPage(appId) + "?activeOnly=true&quantiles=" + quantiles,
+      function(executorMetricsResponse, _ignored_status, _ignored_jqXHR) {
+        var taskMetricKeys = Object.keys(executorMetricsResponse);
+        taskMetricKeys.forEach(function (columnKey) {
+          var row;
+          switch(columnKey) {
+
+            case "JVMHeapMemory":
```

Comment:
Why do we need so many `case` statements here, seems like except for the `default` case all the case statements pretty much doing the same thing?
If you only want to do this for select `columnKey`s, can we just add them to a `Set` and check if it is there and make the call to get the `row` accordingly?

Reply:
How about current?